src/components/RoleItem/RoleItem.tsx

```diff
@@ -0,0 +1,73 @@
+import { Avatar, Typography } from "@mui/material";
+import { UserRoles } from "../../shared/types/UserRolesEnum";
+import { UserType } from "../../shared/types/UserType";
+import "./RoleItem.scss";
+import { useGlobalStore } from "../../state/GlobalStore";
+import MenuButton from "../../shared/UI/MenuButton/MenuButton";
+import { useState } from "react";
+import ChangeRoleModal from "../ChangeRoleModal/ChangeRoleModal";
+
+interface Props {
+  user: UserType;
+  role: RoleType;
+}
+
+const RoleItem = ({ user, role }: Props) => {
+  const { deleteRole, currentRole, currentUserId } = useGlobalStore();
+  const [open, setOpen] = useState(false);
+
+  const getRoleName = () => {
+    switch (role.role) {
+      case UserRoles.Admin:
+        return "Admin";
+      case UserRoles.Owner:
+        return "Owner";
+      default:
+        return "Member";
+    }
+  };
+
+  const getOptions = () => {
+    return [
+      {
+        name: "Kick user",
+        task: () => {
+          deleteRole(role);
+        },
+      },
+      {
+        name: "Change role",
+        task: () => {
+          setOpen(true);
+        },
+      },
+    ];
+  };
+
+  return (
+    <div className="RoleItem">
+      <Avatar
+        src={
+          typeof user !== "undefined" ? user.userPhoto : "../../../profile.jpg"
+        }
+        slotProps={{ img: { referrerPolicy: "no-referrer" } }}
+      />
+      <div className="RoleItem_UserInfo">
+        <Typography variant="h2">{user?.name}</Typography>
+        {getRoleName()}
+      </div>
+      {currentRole !== UserRoles.Member &&
```

Comment:
optional: here you can use same ruleas as in c# - if conditions become too complex, create method with described name such as "isUserCorrect" and use it